File: cabal/packages/numbers-utils/src/Numbers/Utils/Operations/FibonacciUtils.hs
module Numbers.Utils.Operations.FibonacciUtils (fibonacci, fibonacciRandom) where

import System.Random (randomRIO)

fibonacci :: Integer -> Integer
fibonacci n = fibs !! fromIntegral n
    where
        fibs = 0 : 1 : zipWith (+) fibs (tail fibs)

fibonacciRandom :: IO String
fibonacciRandom = do
    randomNum <- randomRIO (1, 50 :: Integer)
    let fib = fibonacci randomNum
    return $ "Fib[" ++ show randomNum ++ "] = " ++ show fib

File: gradle/subprojects/numbers-utils/src/main/kotlin/com/prog/numbers/utils/operations/FibonacciUtils.kt
package com.prog.numbers.utils.operations

import java.math.BigInteger
import kotlin.random.Random

object FibonacciUtils {
    fun fibonacci(n: Int): BigInteger {
        var firstTerm = BigInteger.ZERO
        var secondTerm = BigInteger.ONE

        for (i in 0 until n) {
            val temp = firstTerm
            firstTerm = secondTerm
            secondTerm += temp
        }

        return firstTerm
    }

    fun fibonacciRandom(): String {
        val rng = Random
        val randomNum = rng.nextInt(1, 51)
        val fib = fibonacci(randomNum)

        return "Fib[$randomNum] = $fib"
    }
}
